src/query/union/bitset_union.rs

```diff
@@ -0,0 +1,87 @@
+use std::cell::RefCell;
+
+use crate::docset::DocSet;
+use crate::postings::Postings;
+use crate::query::BitSetDocSet;
+use crate::DocId;
+
+/// Creates a `Posting` that uses the bitset for hits and the docsets for PostingLists.
+///
+/// It is used for the regex phrase query, where we need the union of a large amount of
+/// terms, but need to keep the docsets for the postings.
+pub struct BitSetPostingUnion<TDocSet> {
+    /// The docsets are required to load positions
+    docsets: Vec<RefCell<TDocSet>>,
```

Comment:
Why not 

```suggestion
    docsets: RefCell<Vec<TDocSet>>,
```

Reply:
ah that makes much more sense. The performance difference should be gone with this.